upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -272,6 +275,29 @@ func (tf *TemplateFunctions) GetInstanceGroup(name string) (*kops.InstanceGroup,
 	return ig, nil
 }
 
+// ControlPlaneControllerReplicas returns the amount of replicas for a controllers that should run in the cluster
+// If the cluster has a highly available control plane, this function will return 2, if it has 1 control plane node, it will return 1
+func (tf *TemplateFunctions) ControlPlaneControllerReplicas() int {
+	if tf.HasHighlyAvailableControlPlane() {
+		return 2
+	}
+	return 1
+}
+
+// HasHighlyAvailableControlPlane returns true of the cluster has more than one control plane node. False otherwise.
+func (tf *TemplateFunctions) HasHighlyAvailableControlPlane() bool {
+	cp := 0
+	for _, ig := range tf.InstanceGroups {
+		if ig.Spec.Role == kops.InstanceGroupRoleMaster {
+			cp++
```

Comment:
Perhaps add `*ig.Spec.MinSize` instead if it's non-nil?

Reply:
If it is a CP IG, will that not always be 1?

Reply:
That's how `kops create cluster` provisions them, but I don't think that's a constraint.

Reply:
You are right. One can actually change a master IG to min/maxSize: 2. Which is ... interesting. We don't actually validate that we end up with odd number of masters then. The etcd-manager volume discovery also relies on single instance per IG. 

I think for now, it is safe to say it must be 1 and we lack some validation in this area.